### DISCUSSION TITLE
skipped files/folders in sync_list seem to be ignored

### DISCUSSION THREAD

**Discussion by ittayd** [abraunegg/onedrive#1656] [Q&A] [answered]
I have this sync_list:
```
# Just sync the mediawiki backup folder
/MediawikiBackup

!/MediawikiBackup/data/locks
!/MediawikiBackup/data/*cache.sqlite*
!/MediawikiBackup/data/*jobqueue.sqlite*

```

(Note: I also tried with the skips at the top)

Some files with cache.sqlite suffix are created locally and a locks folder. These are synced to onedrive in the cloud. If I delete them in onedrive in the cloud they are deleted locally.

What I want to achieve is that any file in the MediawikiBackup folder, except those skipped, will be backed up.

**Comment by ittayd** [comment]
Also, instead of looking at just /MediawikiBackup, onedrive scans all other folders in the remote onedrive (IIRC, this was also the case if I used --single-directory), making it slow to start.

**Comment by ittayd** [comment]
Take for example this:
```
Processing MediawikiBackup/data/my_wiki_l10n_cache.sqlite
[DEBUG] skip_file evaluation for: my_wiki_l10n_cache.sqlite
[DEBUG] Evaluation against 'sync_list' for this path: MediawikiBackup/data/my_wiki_l10n_cache.sqlite
[DEBUG] [S]exclude        = false
[DEBUG] [S]excludeMatched = false
[DEBUG] Evaluation against 'sync_list' entry: /MediawikiBackup
[DEBUG] Evaluation against 'sync_list' result: parental path match
[DEBUG] [F]exclude        = false
[DEBUG] [F]excludeMatched = false
[DEBUG] Evaluation against 'sync_list' final result: included for sync
```

**Comment by ittayd** [comment]
I changed the order of entries to put the exclusions first, here is the result:
```
[DEBUG] Checking file: ./MediawikiBackup/data/my_wiki_l10n_cache.sqlite
[DEBUG] skip_file evaluation for: /MediawikiBackup/data/my_wiki_l10n_cache.sqlite
[DEBUG] Evaluation against 'sync_list' for this path: MediawikiBackup/data/my_wiki_l10n_cache.sqlite
[DEBUG] [S]exclude        = false
[DEBUG] [S]excludeMatched = false
[DEBUG] Evaluation against 'sync_list' entry: !/MediawikiBackup/data/locks
[DEBUG] Evaluation against 'sync_list' entry: !/MediawikiBackup/data/*cache.sqlite*
[DEBUG] Evaluation against 'sync_list' result: wildcard pattern matched but must be excluded
[DEBUG] Evaluation against 'sync_list' entry: !/MediawikiBackup/data/*jobqueue.sqlite*
[DEBUG] Evaluation against 'sync_list' entry: /MediawikiBackup
[DEBUG] Evaluation against 'sync_list' result: parental path match
[DEBUG] [F]exclude        = false
[DEBUG] [F]excludeMatched = true
[DEBUG] Evaluation against 'sync_list' final result: included for sync
```

**Comment by ittayd** [comment]
I did another test, to try '/MediawikiBackup/*' and it seems to work

```
[DEBUG] Evaluation against 'sync_list' for this path: MediawikiBackup/data/my_wiki_l10n_cache.sqlite
[DEBUG] [S]exclude        = false
[DEBUG] [S]excludeMatched = false
[DEBUG] Evaluation against 'sync_list' entry: !/MediawikiBackup/data/locks
[DEBUG] Evaluation against 'sync_list' entry: !/MediawikiBackup/data/*cache.sqlite*
[DEBUG] Evaluation against 'sync_list' result: wildcard pattern matched but must be excluded
[DEBUG] Evaluation against 'sync_list' entry: !/MediawikiBackup/data/*jobqueue.sqlite*
[DEBUG] Evaluation against 'sync_list' entry: /MediawikiBackup/*
[DEBUG] Evaluation against 'sync_list' result: wildcard pattern matched but must be excluded
[DEBUG] [F]exclude        = false
[DEBUG] [F]excludeMatched = true
[DEBUG] Evaluation against 'sync_list' final result: EXCLUDED
Skipping item - excluded by sync_list config: MediawikiBackup/data/my_wiki_l10n_cache.sqlite
```

So I think the nuances of parental path and wildcard match should be explained in the usage.

**Comment by abraunegg** [comment]
@ittayd 
What application version are you using?

This sounds like a prior bug that was already fixed.

**Reply by ittayd** [reply]
Built from source yesterday

**Reply by abraunegg** [reply]
OK .. no worries - it would have been good to include this detail, or at least provide an output of your configuration via `--display-config` ...

**Comment by abraunegg** [comment, marked as answer]
@ittayd 
> I have this sync_list:
> ...

As per the [documentation](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#selective-sync-via-sync_list-file) for 'sync_list' usage - it is highly recommended & advised to put exclusions *before* inclusions - which you eventually did.

This would have been the main driver behind the issue you were seeing, but document clarifications are always good & contributions & working examples always welcome.

Also - you can always test your configuration using `--dry-run` before you run your sync, so you test *what* is going to occur and this step is highly recommended.

**Comment by ittayd** [comment]
I did start with the exclusions before inclusions, and used dry-run, but saw it doesn't work. So I switched to putting them after and when that didn't work, wrote this message. But then ran with verbose and looked at the logs where the 'parent' inclusion thing appeared, which made me think. 

Note that the dry run takes the same amount of time as without for going through the files, so for me, since it wasn't syncing a lot of data, it didn't matter much.

And, as mentioned, it looked at all folders in the cloud onedrive. Not clear why.

**Reply by abraunegg** [reply]
@ittayd 
> And, as mentioned, it looked at all folders in the cloud onedrive. Not clear why.

This is because this is how the API works - there is no way to track / get delta changes / object data without getting everything. Post initial sync, things are faster when not using --resync, but when using that, you are back to query all objects again. This is something I cannot solve and is a Microsoft API problem

**Reply by ittayd** [reply]
Not sure, but I think this happened every run
